api/user.go

```diff
@@ -246,3 +246,63 @@ func (a *UserAPI) RemoveUser(c *gin.Context) {
 
 	c.Status(http.StatusNoContent)
 }
+
+type updateUserRequest struct {
+	GitHubToken string `json:"gitHubToken,omitempty"`
+}
+
+// UpdateCurrentUser responds with the authenticated user
+func (a *UserAPI) UpdateCurrentUser(c *gin.Context) {
+	user := auth.GetCurrentUser(c.Request)
+	if user == nil {
+		c.JSON(http.StatusUnauthorized, common.ErrorResponse{
+			Code:    http.StatusUnauthorized,
+			Message: "failed to get current user",
+		})
+		return
+	}
+
+	var updateUserRequest updateUserRequest
+	err := c.BindJSON(&updateUserRequest)
+
+	if err != nil {
+		message := "failed to bind update user request"
+		c.AbortWithStatusJSON(http.StatusBadRequest, common.ErrorResponse{
+			Code:    http.StatusInternalServerError,
+			Message: message,
+			Error:   message,
+		})
+		return
+	}
+
+	db := config.DB()
```

Comment:
db should be injected into the UserAPI struct

Reply:
fixed, thanks!